### ISSUE TITLE
Missing errors/warnings when WordPress-VIP-Go/Minimum is set as standard

### ISSUE THREAD

**Issue by matthewfarlymn** [Automattic/VIP-Coding-Standards#271]
## Bug Description

When WordPress-VIP is set as my PHPCS standard I see errors/warnings for: 

- 'yoda' conditionals
- missing semicolons
- strict comparisons
- correct spacing (i.e. a single space before and after parenthesis)
- equal sign alignment

The list goes on...

If I change my PHPCS standard to use WordPress-VIP-Go or WordPressVIPMinimum none of these errors/warnings are displaying anymore. The only errors/warnings I am seeing are:

- $_POST
- $_GET
- escaping
- nonce 

I have tested my code by making sure these mistakes exist and switching between the coding standards in order to verify PHPCS is working correctly.

## Minimal Code Snippet

Below is my CodeSniffer.conf

```
$phpCodeSnifferConfig = array (
  'installed_paths' => '/Users/matthewfarlymn/Sites/wpcs,/Users/matthewfarlymn/Sites/vipcs',
  'default_standard' => 'WordPress-VIP-Go',
);
```

## Error Code

Running - WordPress-VIP-Go standard
```
FOUND 1 ERROR AND 22 WARNINGS AFFECTING 6 LINES
-------------------------------------------------------------------------------------------------------------------------------------------------------
  260 | ERROR   | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '==='.
  358 | WARNING | Processing form data without nonce verification.
  358 | WARNING | Processing form data without nonce verification.
 1363 | WARNING | Detected access of super global var $_GET, probably needs manual inspection.
 1363 | WARNING | Processing form data without nonce verification.
 1363 | WARNING | Detected access of super global var $_GET, probably needs manual inspection.
 1363 | WARNING | Processing form data without nonce verification.
 1377 | WARNING | Detected access of super global var $_GET, probably needs manual inspection.
 1377 | WARNING | Processing form data without nonce verification.
 1377 | WARNING | Detected access of super global var $_GET, probably needs manual inspection.
 1377 | WARNING | Processing form data without nonce verification.
 1377 | WARNING | Detected access of super global var $_GET, probably needs manual inspection.
 1377 | WARNING | Processing form data without nonce verification.
 1377 | WARNING | Detected access of super global var $_GET, probably needs manual inspection.
 1377 | WARNING | Processing form data without nonce verification.
 1391 | WARNING | Detected access of super global var $_GET, probably needs manual inspection.
 1391 | WARNING | Processing form data without nonce verification.
 1391 | WARNING | Detected access of super global var $_GET, probably needs manual inspection.
 1391 | WARNING | Processing form data without nonce verification.
 1392 | WARNING | Detected access of super global var $_GET, probably needs manual inspection.
 1392 | WARNING | Processing form data without nonce verification.
 1392 | WARNING | Detected access of super global var $_GET, probably needs manual inspection.
 1392 | WARNING | Processing form data without nonce verification.
-------------------------------------------------------------------------------------------------------------------------------------------------------
```

Exact same file running - WordPress-VIP standard
```
FOUND 25 ERRORS AND 38 WARNINGS AFFECTING 31 LINES
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  183 | ERROR   | [x] Line indented incorrectly; expected 2 tabs, found 3
  185 | ERROR   | [x] Line indented incorrectly; expected 2 tabs, found 3
  260 | ERROR   | [ ] All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '==='.
  358 | WARNING | [ ] Processing form data without nonce verification.
  358 | WARNING | [ ] Processing form data without nonce verification.
  809 | ERROR   | [x] Line indented incorrectly; expected 4 tabs, found 3
  813 | ERROR   | [x] Line indented incorrectly; expected at least 4 tabs, found 3
  899 | ERROR   | [x] Expected 1 spaces after opening bracket; 0 found
  903 | ERROR   | [ ] The $text arg must be a single string literal, not "'Subscribe to our '.$title.' newsletter'".
  903 | ERROR   | [x] Concat operator must be surrounded by a single space
  926 | ERROR   | [x] Inline PHP statement must end with a semicolon
  926 | ERROR   | [x] Inline PHP statement must end with a semicolon
 1177 | ERROR   | [ ] Assignments must be the first block of code on a line
 1363 | ERROR   | [x] Opening PHP tag must be on a line by itself
 1363 | ERROR   | [x] No space after opening parenthesis is prohibited
 1363 | WARNING | [ ] Detected access of super global var $_GET, probably needs manual inspection.
 1363 | WARNING | [ ] Processing form data without nonce verification.
 1363 | WARNING | [ ] Found: ==. Use strict comparisons (=== or !==).
 1363 | WARNING | [ ] Detected access of super global var $_GET, probably needs manual inspection.
 1363 | WARNING | [ ] Processing form data without nonce verification.
 1363 | ERROR   | [x] Expected 1 space(s) after closing parenthesis; found 0
 1363 | ERROR   | [x] No space before closing parenthesis is prohibited
 1363 | ERROR   | [x] Space between opening control structure and T_COLON is required
 1378 | WARNING | [ ] Detected access of super global var $_GET, probably needs manual inspection.
 1378 | WARNING | [ ] Processing form data without nonce verification.
 1378 | WARNING | [ ] Detected access of super global var $_GET, probably needs manual inspection.
 1378 | WARNING | [ ] Processing form data without nonce verification.
 1378 | WARNING | [ ] Detected access of super global var $_GET, probably needs manual inspection.
 1378 | WARNING | [ ] Processing form data without nonce verification.
 1378 | WARNING | [ ] Detected access of super global var $_GET, probably needs manual inspection.
 1378 | WARNING | [ ] Processing form data without nonce verification.
 1378 | ERROR   | [x] Expected 1 space(s) after closing parenthesis; found 0
 1378 | ERROR   | [x] Space between opening control structure and T_COLON is required
 1380 | WARNING | [ ] Found precision alignment of 2 spaces.
 1381 | WARNING | [ ] Found precision alignment of 2 spaces.
 1382 | WARNING | [ ] Found precision alignment of 2 spaces.
 1386 | WARNING | [ ] Found precision alignment of 2 spaces.
 1387 | WARNING | [ ] Found precision alignment of 2 spaces.
 1390 | WARNING | [ ] Found precision alignment of 2 spaces.
 1391 | WARNING | [ ] Found precision alignment of 2 spaces.
 1392 | WARNING | [ ] Found precision alignment of 2 spaces.
 1392 | WARNING | [ ] Detected access of super global var $_GET, probably needs manual inspection.
 1392 | WARNING | [ ] Processing form data without nonce verification.
 1392 | WARNING | [ ] Detected access of super global var $_GET, probably needs manual inspection.
 1392 | WARNING | [ ] Processing form data without nonce verification.
 1393 | WARNING | [ ] Found precision alignment of 2 spaces.
 1393 | WARNING | [ ] Detected access of super global var $_GET, probably needs manual inspection.
 1393 | WARNING | [ ] Processing form data without nonce verification.
 1393 | WARNING | [ ] Detected access of super global var $_GET, probably needs manual inspection.
 1393 | WARNING | [ ] Processing form data without nonce verification.
 1395 | WARNING | [ ] Found precision alignment of 2 spaces.
 1414 | ERROR   | [x] Spaces must be used for mid-line alignment; tabs are not allowed
 1414 | WARNING | [x] Equals sign not aligned with surrounding assignments; expected 6 spaces but found 8 spaces
 1420 | WARNING | [ ] A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.
 1422 | ERROR   | [x] Blank line found after control structure
 1424 | ERROR   | [x] Blank line found after control structure
 1433 | ERROR   | [x] Spaces must be used for mid-line alignment; tabs are not allowed
 1433 | WARNING | [x] Equals sign not aligned with surrounding assignments; expected 6 spaces but found 12 spaces
 1439 | WARNING | [ ] A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.
 1439 | ERROR   | [x] Space found before comma in function call
 1441 | ERROR   | [x] Blank line found after control structure
 1459 | WARNING | [ ] A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.
 1461 | ERROR   | [x] Blank line found after control structure
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 24 MARKED SNIFF VIOLATIONS AUTOMATICALLY
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

## Environment

| Question                | Answer
| ------------------------| -------
| PHP version             | 7.1.6 
| PHP_CodeSniffer version | 3.3.0
| WPCS version           | 1.2.0
| VIPCS version           | 0.3.0

## Tested Against `master` branch?

- [ X ] I have verified the issue still exists in the `master` branch of VIPCS.

**Comment by GaryJones** [comment]
TL;DR Forget about WPCS's `WordPress-VIP` - it's not recommended by VIP, nor used by VIP.

The VIP Coding Standards are not concerned with whitespace, yoda conditions, equal sign alignment and other _code style_ considerations. It's more focused on security and performance.

`WordPressVIPMinimum` is the standard for the WPCOM platform. 

`WordPress-VIP-Minimum` is the standard for the VIP Go platform, and is the focus of further improvements.

If you want to also follow the WordPress coding standards, and the VIP Go standards, then it that's possible and fine as well. Include both standards in your `.phpcs.xml.dist`.

I'll close this, as there's no bug here, but feel free to continue if you have further questions.

**Comment by matthewfarlymn** [comment]
Thanks for confirming @GaryJones do you by chance have a reference for running multiple coding standards in a single `phpcs.xml.dist` file?

**Comment by GaryJones** [comment]
Here's my [plugin boilerplate PHPCS config file](https://github.com/GaryJones/plugin-boilerplate/blob/master/.phpcs.xml.dist).

It runs:
- `PHPCompatibilityWP`: checks my code is compatible with PHP 7.1 and above, though I also check that the [main plugin file is PHP 5.2+ compatible](https://github.com/GaryJones/plugin-boilerplate/blob/9eefed64c93be51aafa2a13ed00228105bdaa5b7/composer.json#L54).
- `WordPress-Extra` (which includes `WordPress-Core`): Everything from the WP Handbook and other good practices.
- `NeutronStandard`, which is what one of the A8C teams came up with for PHP 7+ code.

You may well want to swap out `NeutronStandard` for `WordPress-VIP-Go`, but otherwise it's probably a good starting point. Just ensure that your `composer.json` file contains the references to the various standards packages, and has the DealerDirect composer plugin to automatically register the standards in the local version of PHPCS (avoid global PHPCS).
